src/program_iterator.jl

```diff
@@ -49,7 +49,7 @@ macro programiterator(ex)
     generate_iterator(__module__, ex)
 end
 
-function generate_iterator(mod::Module, ex::Expr, mut::Bool=false)
+function generate_iterator(mod::Module, ex::Expr, mut::Bool=true)
```

Comment:
Why do you change the default here?

Reply:
Mutability is needed temporarily. The optional solver argument of a program iterator is set to nothing and then replaced with an actual instance of a Solver inside an iterator.

During the hackathon, we should find a better place to instantiate the default solver. The best place to do this would be in the constructor of the program iterator. In that case, mutability is not needed anymore and the default value of mut can be set to false again.